src/functions.jl

```diff
@@ -138,3 +138,52 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     data = append!(parent(A), unname(B))
     return NamedDimsArray{newL}(data)
 end
+
+################################################
+# map, collect
+
+# no explicit method needed for map(f, ::NamedDimsArray)
+for (T, S) in [
+    (:NamedDimsArray, :AbstractArray),
+    (:AbstractArray, :NamedDimsArray),
+    (:NamedDimsArray, :NamedDimsArray),
+    ]
+    for fun in [:map, :map!]
+
+        @eval function Base.$fun(f, A::$T, B::$S, Cs::AbstractArray...)
```

Comment:
For each function the generated functions are the same. You can create two functions `_map` and `_map!` and create the versions of `map` and `map!` that call the appropriate `_` function. That might be easier on the compiler if it decides it doesn't need to specialize on either of `T` or `S`.

Reply:
I actually don't know if that would have an effect on load time. Up to you whether you want to do that.

Reply:
You could split this, but will it help anything? I thought that what made this package fast was that the compiler specialised not just for every `T, S` but for every combination of names. (But am not certain how one would check this.)

Besides, it's a three-line function repeated 3 times, which doesn't seem like much.